index.js

```diff
@@ -39,25 +33,37 @@ var lnCms = angular.module('lnCms', [
             name: route.state,
             url: route.url,
             templateUrl: 'templates/' + route.template + '/template.html',
+            controller: 'LnViewController as vm',
+            resolve: {
+              staticData: ['lnCmsClientService', function(lnCmsClientService){
+                return lnCmsClientService.getStatic();
+              }],
+              viewData: ['lnCmsClientService', '$stateParams', function(lnCmsClientService, $stateParams){
+                var endpoint = route.endpoint || 'post';
+                var params = route.params || {};
+                angular.merge(params, $stateParams);
+                return lnCmsClientService.getData(endpoint, params);
+              }],
+            },
             data: {
               endpoint: (route.endpoint || 'post'),
               fixedParams: (route.params || {}),
               request: 'request' in route ? route.request : true
             }
```

Comment:
I think this might not be useful any more and in case we want to get the params of the current state we can always inject `$state` to retrieve the data from the current state in any particular component.

Let me know what do you think so I can follow that direction.

Thanks.